healthz/clusterstatus/clusterstatus.go

```diff
@@ -0,0 +1,156 @@
+// Package clusterstatus implements checks and status of
+// monitored K8S cluster
+package clusterstatus
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gravitational/satellite/healthz/config"
+	"github.com/gravitational/satellite/healthz/utils"
+	"github.com/gravitational/trace"
+)
+
+// ClusterStatus represent current status of monitored K8S cluster
+type ClusterStatus struct {
+	K8S struct {
+		Healthy      bool
+		NodesTotal   int
+		NodesHealthy int
+	}
+	Etcd struct {
+		Healthy bool
+	}
+	Mux sync.Mutex `json:"-"`
+}
+
+type key int
+
+const (
+	contextKey                 key = iota
+	statusK8SClusterUnhealthy      = "K8S cluster is unhealthy"
+	statusEtcdClusterUnhealthy     = "Etcd cluster is unhealthy"
+)
+
+// AttachToContext adds *ClusterStatus to context
+func AttachToContext(ctx context.Context, status *ClusterStatus) context.Context {
+	return context.WithValue(ctx, contextKey, status)
+}
+
+// FromContext tries to get *ClusterStatus from context
+func FromContext(ctx context.Context) (*ClusterStatus, bool) {
+	status, ok := ctx.Value(contextKey).(*ClusterStatus)
+	return status, ok
+}
+
+// NewClusterStatus initializes state as healthy cluster
+func NewClusterStatus() *ClusterStatus {
+	c := &ClusterStatus{}
+	c.K8S.Healthy = true
+	c.Etcd.Healthy = true
+	return c
+}
+
+func (c *ClusterStatus) String() string {
+	c.Mux.Lock()
+	defer c.Mux.Unlock()
+	s, err := json.Marshal(c)
+	if err != nil {
+		log.Error(trace.Wrap(err).Error())
+	}
+	return string(s)
+}
+
+func (c *ClusterStatus) updateK8sClusterStatus() error {
+	_, err := utils.ExecCommand("kubectl", "cluster-info")
+	c.Mux.Lock()
+	defer c.Mux.Unlock()
+	c.K8S.Healthy = (err == nil) // No errors -- healthy
+	return nil
+}
+
+func (c *ClusterStatus) updateEtcdStatus() error {
+	_, err := utils.ExecCommand("etcdctl", "cluster-health")
+	c.Mux.Lock()
+	defer c.Mux.Unlock()
+	c.Etcd.Healthy = (err == nil) // No errors -- healthy
+	return nil
+}
+
+func (c *ClusterStatus) updateK8SNodesStatus() error {
+	out, err := utils.ExecCommand("kubectl", "get", "nodes", "--no-headers", "-o=wide")
```

Comment:
you still need to replace this with k8s client call as in my examples on 

https://github.com/gravitational/rigging/blob/master/tool/rig/main.go#L124